courses/machine_learning/intro_to_tensorflow/4_keras_functional_api.ipynb

```diff
@@ -0,0 +1,440 @@
+{
```

Comment:
<p>I don't like that this code depends on the number of examples in the dataset in a hardcoded way. We need to change that variable manually if we switch the dataset (e.g. a smaller dataset may crash the training, which happened to me in the training at scale with the same code pattern). Also, we need to know the exact number of examples ahead of time, and pass it to the code, which adds complexity when we package the code into a container later on)</p><p>Not sure the best way to address that though... any idea?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/576/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Reply:
<p>This is not size of training set. It is total number of training examples to train on. I'll write an article explaining difference and why we do it the way. The crash is a bug that we should fix. The concept is correct, though. </p>